diff --git a/PLACES b/PLACES
index c23bf61..748800e 100644
--- a/PLACES
+++ b/PLACES
@@ -1,5 +1,5 @@
 #
-# Example .tzlist Place list file, as used by slashtime
+# Example tzlist Place list file, as used by slashtime
 #
 # Copyright (c) 2003-2008 Operational Dynamics Consulting Pty Ltd 
 # 
@@ -8,9 +8,9 @@
 # version 2". See the LICENCE file for the terms governing usage and
 # redistribution.
 #
-# =========================================================
-# ** THIS FILE SHOULD BE .tzlist IN YOUR $HOME DIRECTORY **
-# =========================================================
+# ========================================================
+# ** THIS FILE SHOULD BE $HOME/.config/slashtime/tzlist **
+# ========================================================
 # 
 # Purpose
 # -------
@@ -71,6 +71,7 @@
 
 #
 # Now that you've read all this, you can get rid of the comments, change
-# the Places to things that make you happy, and copy the file to ~/.tzlist
+# the Places to things that make you happy, and copy the file to
+# ~/.config/slashtime/tzlist
 #
 # vim: set ts=25:
diff --git a/README.markdown b/README.markdown
index d67e21c..6242930 100644
--- a/README.markdown
+++ b/README.markdown
@@ -17,7 +17,7 @@ For the impatient:
 	$ ./configure
 	$ make
 
-but don't forget to create a `.tzlist` file; see [`PLACES`](PLACES).
+but don't forget to create a `tzlist` file; see [`PLACES`](PLACES).
 
 There is also the old Perl script which is a standalone command line version,
 it doesn't need building, just copying somewhere where binaries live:
@@ -135,12 +135,13 @@ menu.
 Places list
 -----------
 
-The cities shown by slashtime is drawn from the list found in a file called
-.tzlist in your home directory. See the [PLACES](PLACES) file for a fully
+The cities shown by slashtime is drawn from the list found in a file located
+at $HOME/.config/slashtime/tzlist. See the [PLACES](PLACES) file for a fully
 documented example.
 
 Don't whine to me about the lack of UI to change the hard coded Places list. If
-you want your own list of cities, then whip up a `~/.tzlist` file.
+you want your own list of cities, then whip up a `~/.config/slashtime/tzlist`
+file.
 
 
 HISTORY
diff --git a/src/java/slashtime/services/Loader.java b/src/java/slashtime/services/Loader.java
index 213c954..f9c21ad 100644
--- a/src/java/slashtime/services/Loader.java
+++ b/src/java/slashtime/services/Loader.java
@@ -62,7 +62,7 @@ public final class Loader
     }
 
     /**
-     * Attempt to parse ~/.tzlist for Place data. The file format is
+     * Attempt to parse ~/.config/slashtime/tzlist for Place data. The file format is
      * 
      * "zonename" "City" "Country"
      * 
@@ -155,7 +155,7 @@ public final class Loader
      * file. UTC first, as required elsewhere.
      */
     private static Place[] loadFallbackData() {
-        System.err.println("Warning: ~/.tzlist not found. Using fallback Place list instead.");
+        System.err.println("Warning: ~/.config/slashtime/tzlist not found. Using fallback Place list instead.");
 
         return new Place[] {
             new Place("UTC", "Zulu", "Universal Time"),
diff --git a/src/java/slashtime/util/TimeZoneHelper.java b/src/java/slashtime/util/TimeZoneHelper.java
index 4f965ba..975d019 100644
--- a/src/java/slashtime/util/TimeZoneHelper.java
+++ b/src/java/slashtime/util/TimeZoneHelper.java
@@ -180,7 +180,7 @@ public class TimeZoneHelper
                 + "\t# echo \"Europe/Paris\" > /etc/timezone\n\n"
                 + " - set the TZ variable either globally or when running the program:\n\n"
                 + "\t$ TZ=\"Europe/Paris\" ./slashtime\n\n"
-                + "The zone you specify does need to be in your ~/.tzlist file.");
+                + "The zone you specify does need to be in your ~/.config/slashtime/tzlist file.");
 
         return "";
     }
