diff --git a/src/java/slashtime/ui/UserInterface.java b/src/java/slashtime/ui/UserInterface.java
index e5cb40a..8866a6b 100644
--- a/src/java/slashtime/ui/UserInterface.java
+++ b/src/java/slashtime/ui/UserInterface.java
@@ -86,7 +86,7 @@ public class UserInterface
 
         dialog = new AboutDialog();
         dialog.setAuthors(new String[] {
-            "Andrew Cowie <andrew@operationaldynamics.com>",
+            "Andrew Cowie <istathar@gmail.com>",
         });
         dialog.setComments(_("Show the time in various places!"));
         dialog.setCopyright(_("Copyright")
@@ -95,7 +95,7 @@ public class UserInterface
         dialog.setLogo(images.marble);
         dialog.setTranslatorCredits(_("translator-credits"));
         dialog.setVersion(Version.VERSION);
-        dialog.setWebsite("http://research.operationaldynamics.com/projects/slashtime/");
+        dialog.setWebsite("https://github.com/istathar/slashtime/");
         dialog.setWebsiteLabel(_("Website"));
         dialog.setWrapLicense(true);
 
