From 8fc9d313e7d175c3486be50a841ff48a3224fd42 Mon Sep 17 00:00:00 2001
From: Daniele Napolitano <dnax88@gmail.com>
Date: Tue, 14 Sep 2021 15:26:11 +0200
Subject: [PATCH 71/75] Changed hint effect (a bit less evident)

---
 src/graphic_engine.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/graphic_engine.c b/src/graphic_engine.c
index 4e4ecd5..fc117cb 100644
--- a/src/graphic_engine.c
+++ b/src/graphic_engine.c
@@ -117,7 +117,7 @@ gweled_load_pixmaps (void)
 	g_free(filename);
 	gi_sparkle_pixbuf = sge_register_pixbuf (pixbuf, gi_sparkle_pixbuf);
 
-	filename = g_strdup_printf ("gweled/powerglow_%d.png", prefs.tile_size);
+	filename = g_strdup_printf ("gweled/sparkle_%d.png", prefs.tile_size);
 	pixbuf = sge_load_file_to_pixbuf (filename);
 	if (pixbuf == NULL)
 		exit (-1);
-- 
2.33.1

