diff --git a/src/callbacks.c b/src/callbacks.c
index ff9f4bd..3e4c6b9 100644
--- a/src/callbacks.c
+++ b/src/callbacks.c
@@ -148,7 +148,7 @@ on_quit1_activate (GtkMenuItem * menuitem, gpointer user_data)
         else {
             gchar *filename;
 
-            filename = g_strconcat(g_get_user_config_dir(), "/gweled.saved-game", NULL);
+            filename = g_build_filename(g_get_user_config_dir(), "gweled", "saved-game", NULL);
             unlink(filename);
         }
     }
diff --git a/src/main.c b/src/main.c
index f8534b9..a196d80 100644
--- a/src/main.c
+++ b/src/main.c
@@ -77,7 +77,7 @@ void save_preferences(void)
 	FILE *configfile;
 	GError *error = NULL;
 
-	filename = g_strconcat(g_get_user_config_dir(), "/gweled.conf", NULL);
+	filename = g_build_filename(g_get_user_config_dir(), "gweled", "gweled.conf", NULL);
 
     config = g_key_file_new();
 
@@ -108,7 +108,7 @@ void load_preferences(void)
 	GKeyFile *config;
 	GError *error = NULL;
 
-	filename = g_strconcat(g_get_user_config_dir(), "/gweled.conf", NULL);
+	filename = g_build_filename(g_get_user_config_dir(), "gweled", "gweled.conf", NULL);
 
 	config = g_key_file_new();
 	g_key_file_load_from_file(config, filename,
@@ -152,7 +152,7 @@ void save_current_game(void)
 
     game = gweled_get_current_game();
 
-    filename = g_strconcat(g_get_user_config_dir(), "/gweled.saved-game", NULL);
+    filename = g_build_filename(g_get_user_config_dir(), "gweled", "saved-game", NULL);
 
     stream = fopen(filename, "w");
 
@@ -170,7 +170,7 @@ void load_previous_game(void)
     GweledGameState game;
     gint ret;
 
-    filename = g_strconcat(g_get_user_config_dir(), "/gweled.saved-game", NULL);
+    filename = g_build_filename(g_get_user_config_dir(), "gweled", "saved-game", NULL);
 
     stream = fopen(filename, "r");
 
@@ -451,7 +451,7 @@ int main (int argc, char **argv)
 			      BOARD_HEIGHT * prefs.tile_size);
 
     // check for previous saved game
-    filename = g_strconcat(g_get_user_config_dir(), "/gweled.saved-game", NULL);
+    filename = g_build_filename(g_get_user_config_dir(), "gweled", "saved-game", NULL);
 
 	if(g_file_test(filename, G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR)) {
 	    box = gtk_message_dialog_new (GTK_WINDOW (g_main_window),
